internal/internal_nexus_task_poller.go

```diff
@@ -0,0 +1,167 @@
+		numPollerMetric: newNumPollerMetric(params.MetricsHandler, metrics.PollerTypeNexusTask),
+	}
+}
+
+// Poll the nexus task queue and update the num_poller metric
+func (ntp *nexusTaskPoller) pollNexusTaskQueue(ctx context.Context, request *workflowservice.PollNexusTaskQueueRequest) (*workflowservice.PollNexusTaskQueueResponse, error) {
+	ntp.numPollerMetric.increment()
+	defer ntp.numPollerMetric.decrement()
+
+	return ntp.service.PollNexusTaskQueue(ctx, request)
+}
+
+func (ntp *nexusTaskPoller) poll(ctx context.Context) (interface{}, error) {
+	traceLog(func() {
+		ntp.logger.Debug("nexusTaskPoller::Poll")
+	})
+	request := &workflowservice.PollNexusTaskQueueRequest{
+		Namespace: ntp.namespace,
+		TaskQueue: &taskqueuepb.TaskQueue{Name: ntp.taskQueueName, Kind: enumspb.TASK_QUEUE_KIND_NORMAL},
+		Identity:  ntp.identity,
+		WorkerVersionCapabilities: &commonpb.WorkerVersionCapabilities{
+			BuildId:       ntp.workerBuildID,
+			UseVersioning: ntp.useBuildIDVersioning,
+		},
+	}
+
+	response, err := ntp.pollNexusTaskQueue(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+	if response == nil || len(response.TaskToken) == 0 {
+		// No operation info is available on empty poll. Emit using base scope.
+		ntp.metricsHandler.Counter(metrics.NexusPollNoTaskCounter).Inc(1)
+		return nil, nil
+	}
+
+	return response, nil
+}
+
+// PollTask polls a new task
+func (ntp *nexusTaskPoller) PollTask() (any, error) {
+	return ntp.doPoll(ntp.poll)
+}
+
+// ProcessTask processes a new task
+func (ntp *nexusTaskPoller) ProcessTask(task interface{}) error {
+	if ntp.stopping() {
+		return errStop
+	}
+
+	response := task.(*workflowservice.PollNexusTaskQueueResponse)
+	if response.GetRequest() == nil {
+		// We didn't get a request, poll must have timed out.
+		traceLog(func() {
+			ntp.logger.Debug("Empty Nexus poll response")
+		})
+		return nil
+	}
+
+	metricsHandler, handlerErr := ntp.taskHandler.metricsHandlerForTask(response)
+	if handlerErr != nil {
+		// context wasn't propagated to us, use a background context.
+		_, err := ntp.taskHandler.client.WorkflowService().RespondNexusTaskFailed(
+			context.Background(), ntp.taskHandler.fillInFailure(response.TaskToken, handlerErr))
+		return err
+	}
+
+	executionStartTime := time.Now()
+
+	// Schedule-to-start (from the time the request hit the frontend).
+	scheduleToStartLatency := executionStartTime.Sub(response.GetRequest().GetScheduledTime().AsTime())
+	metricsHandler.Timer(metrics.NexusTaskScheduleToStartLatency).Record(scheduleToStartLatency)
+
+	// Process the nexus task.
+	res, failure, err := ntp.taskHandler.Execute(response)
+
+	// Execution latency (in-SDK processing time).
+	metricsHandler.Timer(metrics.NexusTaskExecutionLatency).Record(time.Since(executionStartTime))
+	if err != nil || failure != nil {
+		metricsHandler.Counter(metrics.NexusTaskExecutionFailedCounter).Inc(1)
+	}
+
+	// Drop the task, nothing to report back.
+	// This is only expected due to context deadline errors.
```

Comment:
If this is the case, confirm it. I am a bit concerned other errors will get swallowed.

Reply:
I'll just return the error just in case, thanks!